### PR TITLE
Bundle python27 on packaging

### PR DESCRIPTION
**Pull request by pjessesco**
https://github.com/pjessesco/appleseed_macos_deps_test

Script doesn't work before #2229 merged. This patch allows to copy python binary that includes pyqt4 and sip.

This PR handles issue #2111.

### CODE REVIEW

File: scripts/appleseed.package.py [hunk 1]
```diff
@@ -539,6 +540,15 @@ def add_dependencies_to_stage(self):
         self.copy_qt_resources("QtGui")
         self.copy_qt_framework("QtOpenGL")
 
+    def add_python_to_stage(self):
+        progress("Mac-specific: Adding Python 2.7 to staging directory")
+
+        safe_make_directory("appleseed/python2.7.15")
```
[the remark this answers is not in the document]
Comment (pjessesco) [reply scripts/appleseed.package.py]: No, I changed it.

File: scripts/appleseed.package.configuration.sample.xml [hunk 2]
```diff
@@ -19,7 +19,7 @@
     <!-- The path to the platform runtime files (Windows only). -->
     <platform_runtime_path>C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\redist\x64\Microsoft.VC140.CRT</platform_runtime_path>
 
-    <!-- The path to a Python 2.7 installation. -->
+    <!-- The path to a Python 2.7 installation (In macOS, It should be path/to/macos_deps/python-2.7.15). -->
```
[the remark this answers is not in the document]
Comment (pjessesco) [reply scripts/appleseed.package.configuration.sample.xml]: I assumed `export MACOS_DEPS=/path/to/appleseed_macos_deps_test/` in my dependency test repository. Maybe it is nonsense to mention it here..

File: scripts/appleseed.package.py [hunk 3]
```diff
@@ -539,6 +540,15 @@ def add_dependencies_to_stage(self):
         self.copy_qt_resources("QtGui")
         self.copy_qt_framework("QtOpenGL")
 
+    def add_python_to_stage(self):
+        progress("Mac-specific: Adding Python 2.7 to staging directory")
+
```
[the remark this answers is not in the document]
Comment (pjessesco) [reply scripts/appleseed.package.py]: Done.

File: scripts/appleseed.package.configuration.sample.xml [hunk 4]
```diff
@@ -19,7 +19,7 @@
     <!-- The path to the platform runtime files (Windows only). -->
     <platform_runtime_path>C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\redist\x64\Microsoft.VC140.CRT</platform_runtime_path>
 
-    <!-- The path to a Python 2.7 installation. -->
+    <!-- The path to a Python 2.7 installation (In macOS, It should be path/to/macos_deps/python27). -->
```
[the remark this answers is not in the document]
Comment (pjessesco) [reply scripts/appleseed.package.configuration.sample.xml]: Ok, done.

**Comment by pjessesco** [comment]
Yes, we have to do.

File: src/appleseed.studio/main/main.cpp [hunk 5]
```diff
@@ -120,7 +120,14 @@ namespace
                 bf::canonical(
                     bf::path(Application::get_root_path()) / "python27"
                 ).make_preferred().string();
-#else
+#endif 
+#ifdef __APPLE__
```
Comment (dictoon) [inline src/appleseed.studio/main/main.cpp]: Let's avoid duplicating code:

    #if defined _WIN32 || __APPLE__
        ...
    #else
        ...
    #endif

Comment (pjessesco) [reply src/appleseed.studio/main/main.cpp]: Done.

**Pull request merged**